elastic/kibana
# [Stateful sidenav] Read space solution setting to set the side nav

In this PR I've updated the `navigation` plugin to read the new active space `solution` property and set the side nav accordingly.

**Note**: There isn't yet any UI to set the `solution` on the space, that will come in a following PR
**Note 2**: e2e functional tests will be added when the UI is ready

## How to test

* Add the following in `kibana.dev.yml`

```yml
xpack.cloud.id: "ftr_fake_cloud_id:aGVsbG8uY29tOjQ0MyRFUzEyM2FiYyRrYm4xMjNhYmM="
xpack.cloud.base_url: "https://cloud.elastic.co/"
xpack.cloud.deployment_url: "/deployments/deploymentId"
xpack.cloud_integrations.experiments.enabled: true
xpack.cloud_integrations.experiments.flag_overrides:
  "solutionNavEnabled": true
```

* Create a space with any of the solution using through API (replace <basePath> with your current base path)

```sh
curl -X "POST" "http://localhost:5601/<basePath>/api/spaces/space" \
     -H 'kbn-xsrf: kibana' \
     -H 'Content-Type: text/plain; charset=utf-8' \
     -u 'elastic:changeme' \
     -d $'{
  "name": "search space",
  "id": "search-space",
  "description": "a description",
  "color": "#5c5959",
  "solution": "es", // "es" | "oblt" | "security"
  "disabledFeatures": []
}
'
```

* Refresh the browser and navigate to the newly created space
* You should see the solution side navigation
